zerver/views/streams.py

```diff
@@ -319,8 +319,10 @@ def add_subscriptions_backend(
         if user_profile.realm.is_zephyr_mirror_realm and not all(stream.invite_only for stream in streams):
             return json_error(_("You can only invite other Zephyr mirroring users to invite-only streams."))
         subscribers = set(principal_to_user_profile(user_profile, principal) for principal in principals)
-    else:
+    elif user_profile.realm.is_zephyr_mirror_realm:
```

Comment:
@YJDave: Okay, I don't understand why you are checking for this, could you please explain?

Reply:
I think this change is wrong. The default behaviour is how it should be.

Reply:
This change was added after steve suggestion [here](https://chat.zulip.org/#narrow/stream/backend/subject/stream.20subscription/near/450176)
The reason is that in Zephyr realm if principals length is more than zero it throws error "You can only invite other Zephyr mirroring users to invite-only streams.", but it does assuming that current user is already subscribed.
So we need to check here manually and if there is Zephyr realm then we auto subscribe current user.

Reply:
Okay, that's fine then but add that as a separate commit and don't remove the regular `else:` clause that was there before. We want to keep that. :)